libs/synchronization/include/hpx/synchronization/barrier.hpp

```diff
@@ -15,11 +15,213 @@
+    /// tasks that are handled by multiple threads. - end note ]
+
+    /// Each barrier phase consists of the following steps:
+    ///   - The expected count is decremented by each call to arrive or
+    ///     arrive_and_drop.
+    ///   - When the expected count reaches zero, the phase completion step is
+    ///     run. For the specialization with the default value of the
+    ///     CompletionFunction template parameter, the completion step is run
+    ///     as part of the call to arrive or arrive_and_drop that caused the
+    ///     expected count to reach zero. For other specializations, the
+    ///     completion step is run on one of the threads that arrived at the
+    ///     barrier during the phase.
+    ///   - When the completion step finishes, the expected count is reset to
+    ///     what was specified by the expected argument to the constructor,
+    ///     possibly adjusted by calls to arrive_and_drop, and the next phase
+    ///     starts.
+    ///
+    /// Each phase defines a phase synchronization point. Threads that arrive
+    /// at the barrier during the phase can block on the phase synchronization
+    /// point by calling wait, and will remain blocked until the phase
+    /// completion step is run.
+
+    /// The phase completion step that is executed at the end of each phase has
+    /// the following effects:
+    ///   - Invokes the completion function, equivalent to completion().
+    ///   - Unblocks all threads that are blocked on the phase synchronization
+    ///     point.
+    /// The end of the completion step strongly happens before the returns from
+    /// all calls that were unblocked by the completion step. For
+    /// specializations that do not have the default value of the
+    /// CompletionFunction template parameter, the behavior is undefined if any
+    /// of the barrier object's member functions other than wait are called
+    /// while the completion step is in progress.
+    ///
+    /// Concurrent invocations of the member functions of barrier, other than
+    /// its destructor, do not introduce data races. The member functions
+    /// arrive and arrive_and_drop execute atomically.
+    ///
+    /// CompletionFunction shall meet the Cpp17MoveConstructible (Table 28) and
+    /// Cpp17Destructible (Table 32) requirements.
+    /// std::is_nothrow_invocable_v<CompletionFunction&> shall be true.
+    ///
+    /// The default value of the CompletionFunction template parameter is an
+    /// unspecified type, such that, in addition to satisfying the requirements
+    /// of CompletionFunction, it meets the Cpp17DefaultConstructible
+    /// requirements (Table 27) and completion() has no effects.
+    ///
+    /// barrier::arrival_token is an unspecified type, such that it meets the
+    /// Cpp17MoveConstructible (Table 28), Cpp17MoveAssignable (Table 30), and
+    /// Cpp17Destructible (Table 32) requirements.
+
```

Comment:
Do you know if doxygen will take the docs above into account with the empty line (same with the other empty lines further up)?

Reply:
I shouldn't have marked all the standardeese as doxygen comments in the first place.